### PR TITLE
Fixes twitch panel

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/3590
Resolves https://github.com/brave/brave-browser/issues/4680
Resolves https://github.com/brave/brave-browser/issues/3417

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
- enable rewards
- go to https://twitch.tv/ilikeblueturtles
- open panel and make sure that panel shows verified check, correct name and image
- go to live streams and vods
- make sure that correct data is displayed in the panel

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [x] Adequate test coverage exists to prevent regressions 
- [x] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: vendor/bat-native-ledger/src/bat/ledger/internal/media/twitch.cc [hunk 1]
```diff
@@ -331,36 +324,37 @@ void MediaTwitch::ProcessMedia(const std::map<std::string, std::string>& parts,
...
+    return;
+  }
+
+  std::string media_id = GetMediaIdFromUrl(visit_data.url,
+                                           publisher_blob);
+
+  std::transform(media_id.begin(),
+                 media_id.end(),
+                 media_id.begin(),
+                 ::tolower);
```
Comment (emerick) [inline vendor/bat-native-ledger/src/bat/ledger/internal/media/twitch.cc]: Can we use `base::ToLowerASCII` here?

File: vendor/bat-native-ledger/src/bat/ledger/internal/media/twitch.cc [hunk 2]
```diff
@@ -400,99 +390,82 @@ void MediaTwitch::OnMediaPublisherInfo(
...
 
-    if (media_id.find("_vod_") != std::string::npos) {
-      // VOD
-      std::vector<std::string> media_props =
-          braveledger_bat_helper::split(media_id, MEDIA_DELIMITER);
-      if (media_props.empty()) {
-        return;
-      }
+  ledger::TwitchEventInfo old_event;
+  std::map<std::string, ledger::TwitchEventInfo>::const_iterator iter =
```
Comment (emerick) [inline vendor/bat-native-ledger/src/bat/ledger/internal/media/twitch.cc]: Nit: Could use `auto` here to make this declaration more concise.

File: vendor/bat-native-ledger/src/bat/ledger/internal/media/twitch.cc [hunk 3]
```diff
@@ -647,17 +614,22 @@ void MediaTwitch::SavePublisherInfo(const uint64_t duration,
...
 
-  std::string publisher_id = GetPublisherKey(channel_id);
-  std::string url = publisher_url + "/videos";
+  std::string key = publisher_key;
+  if (key.empty()) {
+    key = GetPublisherKey(channel_id);
+  }
 
-  if (publisher_id.empty()) {
+  const std::string url = publisher_url + "/videos";
```
Comment (emerick) [inline vendor/bat-native-ledger/src/bat/ledger/internal/media/twitch.cc]: It seems like `url` isn't used anywhere, unless I'm just missing it.

File: vendor/bat-native-ledger/src/bat/ledger/internal/media/twitch.cc [hunk 4]
```diff
@@ -251,12 +245,11 @@ std::string MediaTwitch::GetFaviconUrl(
     return std::string();
   }
 
-  return braveledger_media::ExtractData(publisher_blob,
-    "<figure class=\"tw-avatar tw-avatar--size-36\">"
-    "<div class=\"tw-border-radius-medium tw-overflow-hidden\">"
-    "<img class=\"tw-avatar__img tw-image\" alt=\"" + handle + "\" "
-    "src=\"",
-    "\"");
+  const std::string wrapper = braveledger_media::ExtractData(publisher_blob,
```
Comment (jasonrsadler) [inline vendor/bat-native-ledger/src/bat/ledger/internal/media/twitch.cc]: This won't work if the user is logged in.

The first instance of this pattern when a user is logged in appears for the logged in user's avatar

When not logged in:
UNADJUSTEDNONRAW_mini_2a9


When logged in with my account it shows my avatar:
UNADJUSTEDNONRAW_mini_2a8

File: vendor/bat-native-ledger/src/bat/ledger/internal/media/twitch.cc [hunk 5]
```diff
@@ -400,99 +390,82 @@ void MediaTwitch::OnMediaPublisherInfo(
...
   }
+
+  // Live stream
+  SavePublisherInfo(real_duration,
+                    media_key,
+                    GetMediaURL(media_id),
+                    media_id,
+                    visit_data,
+                    window_id,
+                    "",
```
Comment (jasonrsadler) [inline vendor/bat-native-ledger/src/bat/ledger/internal/media/twitch.cc]: Can we please use `std::string()` or `{}`?

`""` still requires significant processing that can't be optimized away. Insignificant, but not if we start doing this all over the place.

File: vendor/bat-native-ledger/src/bat/ledger/internal/media/twitch.cc (hunk 3 above)
Comment (jasonrsadler) [reply vendor/bat-native-ledger/src/bat/ledger/internal/media/twitch.cc]: We're good on this:
https://github.com/brave/brave-core/pull/2618/files#diff-410ec7b37fbed59610d3617cce571df3R646

it's used outside of diff

File: vendor/bat-native-ledger/src/bat/ledger/internal/media/twitch.cc (hunk 2 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/media/twitch.cc]: done

File: vendor/bat-native-ledger/src/bat/ledger/internal/media/twitch.cc (hunk 3 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/media/twitch.cc]: fixed

File: vendor/bat-native-ledger/src/bat/ledger/internal/media/twitch.cc (hunk 1 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/media/twitch.cc]: done

File: vendor/bat-native-ledger/src/bat/ledger/internal/media/twitch.cc (hunk 5 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/media/twitch.cc]: done

File: vendor/bat-native-ledger/src/bat/ledger/internal/media/twitch.cc (hunk 4 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/media/twitch.cc]: done

**Comment by kjozwiak** [comment]
Quickly went through the test case outlined under https://github.com/brave/brave-core/pull/2618#issue-285655781 to make sure things are working before uplifting https://github.com/brave/brave-core/pull/2633. Looks like things are generally working. QA will eventually go through a more detailed check/verification. Example of the fix working under `0.68.35 Chromium: 75.0.3770.80`:

Screen Shot 2019-06-11 at 10 19 09 AM
Screen Shot 2019-06-11 at 10 15 48 AM
Screen Shot 2019-06-11 at 10 11 57 AM

**Pull request merged**